drivers/uart/CMakeLists.txt

```
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+zephyr_sources_ifdef(CONFIG_NRF_LPUART uart_nrf_lpuart.c)
```

This `add_subdirectory_ifdef(CONFIG_NRF_LPUART uart)` secures that this CMakeLists.txt file is only included when `CONFIG_NRF_LPUART=y`, so no need to recheck here.
 https://github.com/nrfconnect/sdk-nrf/blob/0ffd8519a086608b6ac580a8aaf1878d1a693b04/drivers/CMakeLists.txt#L11

But a different question, why is this located in `drivers/uart` ?
Zephyr uses: [`drivers/serial/`](https://github.com/nrfconnect/sdk-zephyr/tree/master/drivers/serial) so I would request to rename the folder: `uart --> serial` and then use:
```suggestion
zephyr_library_amend()
zephyr_library_sources(uart_nrf_lpuart.c)
```
instead.

---

Seems `/drivers/serial/` is not the right location, see: https://github.com/nrfconnect/sdk-nrf/pull/2591#pullrequestreview-480632959

---

done

---

on the other hand it implements API. In zephyr `drivers/serial` there is `uart_rtt.c` which implements API and have no compliance with UART physical transport.

---

Yes, so perhaps `uart_nrf_sw_lpuart.c` is acceptable. And yes, all descriptions of it must clearly state that it is not a standard UART driver replacement, to avoid any confusion in this regard.

---

renamed. Kconfig help states:
```
 Low power UART implements UART API and extends standard UART communication with 2 pins protocol for receiver wake up and flow control.
```